contracts/tokens/NFToken.sol

```diff
@@ -186,7 +146,41 @@ contract NFToken is
   )
     external
   {
-    _safeTransferFrom(_from, _to, _tokenId, _data);
+    // valid NFT check
+    require(_from != address(0));
+    require(idToOwner[_tokenId] == _from);
+    require(_to != address(0));
+
+    // can transfer
+    require(
+      _from == msg.sender
+      || idToApprovals[_tokenId] == msg.sender
+      || ownerToOperators[_from][msg.sender]
+    );
```

Comment:
Is there a GAS  difference between this one requirement vs multiple ones (one per ||) ?